packages/client/src/monaco-language-client.ts

```diff
@@ -5,37 +5,37 @@
-import { InlayHintsFeature } from 'vscode-languageclient/lib/common/inlayHint';
-import { DiagnosticFeature } from 'vscode-languageclient/lib/common/diagnostic';
-import { ProgressFeature } from 'vscode-languageclient/lib/common/progress';
+} from 'vscode-languageclient/lib/common/client.js';
+import { ConfigurationFeature, SyncConfigurationFeature } from 'vscode-languageclient/lib/common/configuration.js';
+import { DidChangeTextDocumentFeature, DidCloseTextDocumentFeature, DidOpenTextDocumentFeature, DidSaveTextDocumentFeature, WillSaveFeature, WillSaveWaitUntilFeature } from 'vscode-languageclient/lib/common/textSynchronization.js';
+import { CompletionItemFeature } from 'vscode-languageclient/lib/common/completion.js';
+import { HoverFeature } from 'vscode-languageclient/lib/common/hover.js';
+import { SignatureHelpFeature } from 'vscode-languageclient/lib/common/signatureHelp.js';
+import { DefinitionFeature } from 'vscode-languageclient/lib/common/definition.js';
+import { ReferencesFeature } from 'vscode-languageclient/lib/common/reference.js';
+import { DocumentHighlightFeature } from 'vscode-languageclient/lib/common/documentHighlight.js';
+import { DocumentSymbolFeature } from 'vscode-languageclient/lib/common/documentSymbol.js';
+import { CodeActionFeature } from 'vscode-languageclient/lib/common/codeAction.js';
+import { CodeLensFeature } from 'vscode-languageclient/lib/common/codeLens.js';
+import { DocumentFormattingFeature, DocumentOnTypeFormattingFeature, DocumentRangeFormattingFeature } from 'vscode-languageclient/lib/common/formatting.js';
+import { RenameFeature } from 'vscode-languageclient/lib/common/rename.js';
+import { DocumentLinkFeature } from 'vscode-languageclient/lib/common/documentLink.js';
+import { ExecuteCommandFeature } from 'vscode-languageclient/lib/common/executeCommand.js';
+import { TypeDefinitionFeature } from 'vscode-languageclient/lib/common/typeDefinition.js';
+import { ImplementationFeature } from 'vscode-languageclient/lib/common/implementation.js';
+import { ColorProviderFeature } from 'vscode-languageclient/lib/common/colorProvider.js';
+import { WorkspaceFoldersFeature } from 'vscode-languageclient/lib/common/workspaceFolder.js';
+import { FoldingRangeFeature } from 'vscode-languageclient/lib/common/foldingRange.js';
+import { DeclarationFeature } from 'vscode-languageclient/lib/common/declaration.js';
+import { SelectionRangeFeature } from 'vscode-languageclient/lib/common/selectionRange.js';
+import { SemanticTokensFeature } from 'vscode-languageclient/lib/common/semanticTokens.js';
+import { LinkedEditingFeature } from 'vscode-languageclient/lib/common/linkedEditingRange.js';
+import { InlayHintsFeature } from 'vscode-languageclient/lib/common/inlayHint.js';
+import { DiagnosticFeature } from 'vscode-languageclient/lib/common/diagnostic.js';
+import { ProgressFeature } from 'vscode-languageclient/lib/common/progress.js';
```

Comment:
Did you have time to take a look at this:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#ecmascript-module-support-in-nodejs
If you decide to go the route of type=module and use
```
"compilerOptions": {
    "module": "node16"
}
```
then you need to use this syntax. This basically aligns the JS/TS syntax here.

Reply:
(Btw, the whole section is about ESM in Nodejs, while the client package will be used in the browser)

Reply:
Yes, true, but this way it could be used in a node context (if this is required is another question) and the browser context. The package spec and the code match. This was my driver.
Usage in the browser is possible as before. I have tested this with the pre-releases.

Reply:
... and I want to get rid of CJS. People can transform their code using bundlers, but here the way is hopefully clear for the future 🙈

Reply:
I strongly agree, is there anything similar that should be done on monaco-vscode-api?

Reply:
You already wrote ESM code from day one, so there is no issue, I think. Here, there where still a couple of places where I changed the syntax (independent of file extension) and it became obvious when I restricted the TS compiler to use `module=node16`.
I honestly have no good answer in where the overlap is in between ES20xx/ES2xxx and Node16/NodeNext is. Need to investigate that.